### PR TITLE
Add create_projection_pdf

### PR DESCRIPTION
**Pull request by apuignav**
To start reviewing, tests still missing.

### CODE REVIEW

**Comment by apuignav** [comment]
OK, tests are failing with

```
E                   ValueError: TODO observables not contained
```

Looks like I hit a TODO?

File: zfit/core/basepdf.py [hunk 1]
```diff
@@ -479,6 +479,14 @@ def get_yield(self) -> Union[Parameter, None]:
         #     raise zexception.ExtendedPDFError("PDF is not extended, cannot get yield.")
         return self._yield
 
+    def create_projection_pdf(self, x: ztyping.XType, limits_to_integrate: ztyping.LimitsType) -> 'BasePDF':
+        """Create a PDF projecting out some of the dimensions."""
+        from ..models.special import SimplePDF
+        SimplePDF(obs=self.obs.get_subspace(obs=[obs for obs in self.obs.obs
+                                                 if obs not in limits_to_integrate.obs]),
+                  func=self.partial_integrate(x,
```
[the remark this answers is not in the document]
Comment (apuignav) [reply zfit/core/basepdf.py]: True, I fix it.

[the remark this answers is not in the document]
Comment (mayou36) [reply zfit/core/basepdf.py]: yes, I would. Basically the same input check as in `partial_integrate`

[the remark this answers is not in the document]
Comment (apuignav) [reply zfit/core/basepdf.py]: What I did was to defer to `partial_integrate`, sounds like a plan?

**Comment by apuignav** [comment]
OK, fixed test (too simple?) and code, ready from my side.

File: zfit/models/functor.py [hunk 2]
```diff
@@ -314,8 +314,8 @@ def __init__(self, pdfs: List[ZfitPDF], obs: ztyping.ObsTypeInput = None, name="
     def _unnormalized_pdf(self, x: ztyping.XType):
 
         norm_range = self._get_component_norm_range()
-        return ztf.reduce_prod([pdf.unnormalized_pdf(x, component_norm_range=norm_range.get_subspace(obs=pdf.obs))
-                                for pdf in self.pdfs], axis=0)
+        return np.prod([pdf.unnormalized_pdf(x, component_norm_range=norm_range.get_subspace(obs=pdf.obs))
```
Comment (apuignav) [inline zfit/models/functor.py]: Interesting use of `numpy`. Why is `tensorflow` not good here? It's because we already have numbers?

Comment (mayou36) [reply zfit/models/functor.py]: Yes, it's to use the overloaded `__mul__` operator. The probs are still tensors, but TensorFlow does not offer a "multiply n tensors", only the "reduce_prod", which requires though a fixed axes (but we wan't things to broadcast here)

So it's essentially just a * b * c...

Broadcast from (n, 1) * (1, m) -> (n, m) is needed

(tf.einsum would work as well)

**Comment by mayou36** [comment]
So it's working now, I've also put an example in the tests.

Slightly hacky with the functor norm_range, but it's an implementation detail.

**Comment by apuignav** [comment]
The example is quite nice! We can merge this then?

**Comment by mayou36** [comment]
There we go!

**Pull request merged**